lib/ansible/modules/windows/win_shortcut.ps1

```diff
@@ -0,0 +1,95 @@
+#!powershell
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+$ErrorActionPreference = "Stop"
+$params = Parse-Args $args
+$State = Get-AnsibleParam -obj $params -name "state" -failifempty $true
```

Comment:
I think it would be best if you used 
$State = Get-AnsibleParam -obj $params -name "state"  -default "Present" -ValidateSet "present","absent" -resultobj $resultobj -failifempty $true
then you can get rid of L86 to L89 and if 'state' isn't supplied at all, the default would be to imply state=present